modules/nodesale/nodesale_test.go

```diff
@@ -0,0 +1,114 @@
+package nodesale
+
+import (
+	"context"
+	"flag"
+	"testing"
+	"time"
+
+	"github.com/btcsuite/btcd/chaincfg/chainhash"
+	"github.com/btcsuite/btcd/txscript"
+	"github.com/decred/dcrd/dcrec/secp256k1/v4"
+	"github.com/gaze-network/indexer-network/common"
+	"github.com/gaze-network/indexer-network/core/types"
+	"github.com/gaze-network/indexer-network/internal/postgres"
+	"github.com/gaze-network/indexer-network/modules/nodesale/datagateway"
+	"github.com/gaze-network/indexer-network/modules/nodesale/protobuf"
+	repository "github.com/gaze-network/indexer-network/modules/nodesale/repository/postgres"
+	"github.com/gaze-network/indexer-network/modules/nodesale/repository/postgres/gen"
+	"google.golang.org/protobuf/encoding/protojson"
+	"google.golang.org/protobuf/proto"
+)
+
+var p *Processor
+
+var postgresConf postgres.Config = postgres.Config{
+	User:     "postgres",
+	Password: "P@ssw0rd",
+	DBName:   "gaze_indexer",
+}
+
```

Comment:
there shouldn't be postgres config in test since it shouldn't directly connect to database if it only contains unit tests